{"name": "Проверка отображения заказов пользователя из истории заказов в ленте заказов", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.MoveTargetOutOfBoundsException: Message: Move target (164, 774) is out of bounds of viewport dimensions (1280, 650)\nStacktrace:\nRemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nWebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:191:5\nMoveTargetOutOfBoundsError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:477:5\nassertInViewPort@chrome://remote/content/shared/webdriver/Actions.sys.mjs:2371:11\ndispatch@chrome://remote/content/shared/webdriver/Actions.sys.mjs:1152:21\ndispatch/pendingEvents<@chrome://remote/content/shared/webdriver/Actions.sys.mjs:2040:14\ndispatch@chrome://remote/content/shared/webdriver/Actions.sys.mjs:2039:39\ndispatch/chainEvents<@chrome://remote/content/shared/webdriver/Actions.sys.mjs:1950:27\ndispatch@chrome://remote/content/shared/webdriver/Actions.sys.mjs:1952:7\nperformActions@chrome://remote/content/marionette/actors/MarionetteCommandsChild.sys.mjs:503:23\nreceiveMessage@chrome://remote/content/marionette/actors/MarionetteCommandsChild.sys.mjs:143:31", "trace": "self = <tests.test_order_tape.TestOrderTape object at 0x1071c29c0>, prepare_user = <function prepare_user at 0x106d3ba60>, login_user = None\ndriver = <selenium.webdriver.firefox.webdriver.WebDriver (session=\"f3f50b77-7029-4343-a17a-34daf73635ce\")>\n\n    @allure.title('Проверка отображения заказов пользователя из истории заказов в ленте заказов')\n    @allure.description('1.Залогиниться под пользователем.'\n                        '2.Создать заказ.'\n                        '3.Получить номер заказа из раздела «История заказов».'\n                        '4.Проверить, что данный номер заказа отображается в разделе «Лента заказов».')\n    def test_client_can_find_order_in_history_and_in_tape(self, prepare_user, login_user, driver):\n        main = MainPage(driver)\n>       main.create_order()\n\ntests/test_order_tape.py:28: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/main_page.py:91: in create_order\n    self.drag_and_drop_ingredient()\npages/main_page.py:62: in drag_and_drop_ingredient\n    return self.drag_and_drop(MainLocators.INGREDIENT, MainLocators.ORDER_BASKET)\npages/base_page.py:29: in drag_and_drop\n    self.find_element_located(locator_put)).perform()\n../venv/Diplom__3/lib/python3.12/site-packages/selenium/webdriver/common/action_chains.py:94: in perform\n    self.w3c_actions.perform()\n../venv/Diplom__3/lib/python3.12/site-packages/selenium/webdriver/common/actions/action_builder.py:96: in perform\n    self.driver.execute(Command.W3C_ACTIONS, enc)\n../venv/Diplom__3/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py:347: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x1072a6330>\nresponse = {'status': 500, 'value': '{\"value\":{\"error\":\"move target out of bounds\",\"message\":\"Move target (164, 774) is out of bo...ys.mjs:503:23\\\\nreceiveMessage@chrome://remote/content/marionette/actors/MarionetteCommandsChild.sys.mjs:143:31\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.MoveTargetOutOfBoundsException: Message: Move target (164, 774) is out of bounds of viewport dimensions (1280, 650)\nE       Stacktrace:\nE       RemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nE       WebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:191:5\nE       MoveTargetOutOfBoundsError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:477:5\nE       assertInViewPort@chrome://remote/content/shared/webdriver/Actions.sys.mjs:2371:11\nE       dispatch@chrome://remote/content/shared/webdriver/Actions.sys.mjs:1152:21\nE       dispatch/pendingEvents<@chrome://remote/content/shared/webdriver/Actions.sys.mjs:2040:14\nE       dispatch@chrome://remote/content/shared/webdriver/Actions.sys.mjs:2039:39\nE       dispatch/chainEvents<@chrome://remote/content/shared/webdriver/Actions.sys.mjs:1950:27\nE       dispatch@chrome://remote/content/shared/webdriver/Actions.sys.mjs:1952:7\nE       performActions@chrome://remote/content/marionette/actors/MarionetteCommandsChild.sys.mjs:503:23\nE       receiveMessage@chrome://remote/content/marionette/actors/MarionetteCommandsChild.sys.mjs:143:31\n\n../venv/Diplom__3/lib/python3.12/site-packages/selenium/webdriver/remote/errorhandler.py:229: MoveTargetOutOfBoundsException"}, "description": "1.Залогиниться под пользователем.2.Создать заказ.3.Получить номер заказа из раздела «История заказов».4.Проверить, что данный номер заказа отображается в разделе «Лента заказов».", "steps": [{"name": "Создаем заказ", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.MoveTargetOutOfBoundsException: Message: Move target (164, 774) is out of bounds of viewport dimensions (1280, 650)\nStacktrace:\nRemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nWebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:191:5\nMoveTargetOutOfBoundsError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:477:5\nassertInViewPort@chrome://remote/content/shared/webdriver/Actions.sys.mjs:2371:11\ndispatch@chrome://remote/content/shared/webdriver/Actions.sys.mjs:1152:21\ndispatch/pendingEvents<@chrome://remote/content/shared/webdriver/Actions.sys.mjs:2040:14\ndispatch@chrome://remote/content/shared/webdriver/Actions.sys.mjs:2039:39\ndispatch/chainEvents<@chrome://remote/content/shared/webdriver/Actions.sys.mjs:1950:27\ndispatch@chrome://remote/content/shared/webdriver/Actions.sys.mjs:1952:7\nperformActions@chrome://remote/content/marionette/actors/MarionetteCommandsChild.sys.mjs:503:23\nreceiveMessage@chrome://remote/content/marionette/actors/MarionetteCommandsChild.sys.mjs:143:31\n\n", "trace": "  File \"/Users/ilyabolotskikh/PycharmProjects/venv/Diplom__3/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/Users/ilyabolotskikh/PycharmProjects/Diplom__3/pages/main_page.py\", line 91, in create_order\n    self.drag_and_drop_ingredient()\n  File \"/Users/ilyabolotskikh/PycharmProjects/venv/Diplom__3/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/Users/ilyabolotskikh/PycharmProjects/Diplom__3/pages/main_page.py\", line 62, in drag_and_drop_ingredient\n    return self.drag_and_drop(MainLocators.INGREDIENT, MainLocators.ORDER_BASKET)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/ilyabolotskikh/PycharmProjects/venv/Diplom__3/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/Users/ilyabolotskikh/PycharmProjects/Diplom__3/pages/base_page.py\", line 29, in drag_and_drop\n    self.find_element_located(locator_put)).perform()\n                                            ^^^^^^^^^\n  File \"/Users/ilyabolotskikh/PycharmProjects/venv/Diplom__3/lib/python3.12/site-packages/selenium/webdriver/common/action_chains.py\", line 94, in perform\n    self.w3c_actions.perform()\n  File \"/Users/ilyabolotskikh/PycharmProjects/venv/Diplom__3/lib/python3.12/site-packages/selenium/webdriver/common/actions/action_builder.py\", line 96, in perform\n    self.driver.execute(Command.W3C_ACTIONS, enc)\n  File \"/Users/ilyabolotskikh/PycharmProjects/venv/Diplom__3/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py\", line 347, in execute\n    self.error_handler.check_response(response)\n  File \"/Users/ilyabolotskikh/PycharmProjects/venv/Diplom__3/lib/python3.12/site-packages/selenium/webdriver/remote/errorhandler.py\", line 229, in check_response\n    raise exception_class(message, screen, stacktrace)\n"}, "steps": [{"name": "Загрузка главной страницы приложения Stellar Burgers", "status": "passed", "steps": [{"name": "Находим элемент", "status": "passed", "parameters": [{"name": "locator", "value": "('xpath', \".//main[@class = 'App_componentContainer__2JC2W']\")"}, {"name": "time", "value": "20"}], "start": 1709458420918, "stop": 1709458420919}], "start": 1709458420917, "stop": 1709458420919}, {"name": "Добавляем ингредиент в заказ", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.MoveTargetOutOfBoundsException: Message: Move target (164, 774) is out of bounds of viewport dimensions (1280, 650)\nStacktrace:\nRemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nWebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:191:5\nMoveTargetOutOfBoundsError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:477:5\nassertInViewPort@chrome://remote/content/shared/webdriver/Actions.sys.mjs:2371:11\ndispatch@chrome://remote/content/shared/webdriver/Actions.sys.mjs:1152:21\ndispatch/pendingEvents<@chrome://remote/content/shared/webdriver/Actions.sys.mjs:2040:14\ndispatch@chrome://remote/content/shared/webdriver/Actions.sys.mjs:2039:39\ndispatch/chainEvents<@chrome://remote/content/shared/webdriver/Actions.sys.mjs:1950:27\ndispatch@chrome://remote/content/shared/webdriver/Actions.sys.mjs:1952:7\nperformActions@chrome://remote/content/marionette/actors/MarionetteCommandsChild.sys.mjs:503:23\nreceiveMessage@chrome://remote/content/marionette/actors/MarionetteCommandsChild.sys.mjs:143:31\n\n", "trace": "  File \"/Users/ilyabolotskikh/PycharmProjects/venv/Diplom__3/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/Users/ilyabolotskikh/PycharmProjects/Diplom__3/pages/main_page.py\", line 62, in drag_and_drop_ingredient\n    return self.drag_and_drop(MainLocators.INGREDIENT, MainLocators.ORDER_BASKET)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/ilyabolotskikh/PycharmProjects/venv/Diplom__3/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/Users/ilyabolotskikh/PycharmProjects/Diplom__3/pages/base_page.py\", line 29, in drag_and_drop\n    self.find_element_located(locator_put)).perform()\n                                            ^^^^^^^^^\n  File \"/Users/ilyabolotskikh/PycharmProjects/venv/Diplom__3/lib/python3.12/site-packages/selenium/webdriver/common/action_chains.py\", line 94, in perform\n    self.w3c_actions.perform()\n  File \"/Users/ilyabolotskikh/PycharmProjects/venv/Diplom__3/lib/python3.12/site-packages/selenium/webdriver/common/actions/action_builder.py\", line 96, in perform\n    self.driver.execute(Command.W3C_ACTIONS, enc)\n  File \"/Users/ilyabolotskikh/PycharmProjects/venv/Diplom__3/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py\", line 347, in execute\n    self.error_handler.check_response(response)\n  File \"/Users/ilyabolotskikh/PycharmProjects/venv/Diplom__3/lib/python3.12/site-packages/selenium/webdriver/remote/errorhandler.py\", line 229, in check_response\n    raise exception_class(message, screen, stacktrace)\n"}, "steps": [{"name": "Добавляем ингредиент в заказ", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.MoveTargetOutOfBoundsException: Message: Move target (164, 774) is out of bounds of viewport dimensions (1280, 650)\nStacktrace:\nRemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nWebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:191:5\nMoveTargetOutOfBoundsError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:477:5\nassertInViewPort@chrome://remote/content/shared/webdriver/Actions.sys.mjs:2371:11\ndispatch@chrome://remote/content/shared/webdriver/Actions.sys.mjs:1152:21\ndispatch/pendingEvents<@chrome://remote/content/shared/webdriver/Actions.sys.mjs:2040:14\ndispatch@chrome://remote/content/shared/webdriver/Actions.sys.mjs:2039:39\ndispatch/chainEvents<@chrome://remote/content/shared/webdriver/Actions.sys.mjs:1950:27\ndispatch@chrome://remote/content/shared/webdriver/Actions.sys.mjs:1952:7\nperformActions@chrome://remote/content/marionette/actors/MarionetteCommandsChild.sys.mjs:503:23\nreceiveMessage@chrome://remote/content/marionette/actors/MarionetteCommandsChild.sys.mjs:143:31\n\n", "trace": "  File \"/Users/ilyabolotskikh/PycharmProjects/venv/Diplom__3/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/Users/ilyabolotskikh/PycharmProjects/Diplom__3/pages/base_page.py\", line 29, in drag_and_drop\n    self.find_element_located(locator_put)).perform()\n                                            ^^^^^^^^^\n  File \"/Users/ilyabolotskikh/PycharmProjects/venv/Diplom__3/lib/python3.12/site-packages/selenium/webdriver/common/action_chains.py\", line 94, in perform\n    self.w3c_actions.perform()\n  File \"/Users/ilyabolotskikh/PycharmProjects/venv/Diplom__3/lib/python3.12/site-packages/selenium/webdriver/common/actions/action_builder.py\", line 96, in perform\n    self.driver.execute(Command.W3C_ACTIONS, enc)\n  File \"/Users/ilyabolotskikh/PycharmProjects/venv/Diplom__3/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py\", line 347, in execute\n    self.error_handler.check_response(response)\n  File \"/Users/ilyabolotskikh/PycharmProjects/venv/Diplom__3/lib/python3.12/site-packages/selenium/webdriver/remote/errorhandler.py\", line 229, in check_response\n    raise exception_class(message, screen, stacktrace)\n"}, "steps": [{"name": "Находим элемент", "status": "passed", "parameters": [{"name": "locator", "value": "('xpath', \".//img[@alt = 'Мясо бессмертных моллюсков Protostomia']\")"}, {"name": "time", "value": "20"}], "start": 1709458420919, "stop": 1709458421433}, {"name": "Находим элемент", "status": "passed", "parameters": [{"name": "locator", "value": "('xpath', \".//ul[@class = 'BurgerConstructor_basket__list__l9dp_']\")"}, {"name": "time", "value": "20"}], "start": 1709458421433, "stop": 1709458421436}], "parameters": [{"name": "locator_take", "value": "('xpath', \".//img[@alt = 'Мясо бессмертных моллюсков Protostomia']\")"}, {"name": "locator_put", "value": "('xpath', \".//ul[@class = 'BurgerConstructor_basket__list__l9dp_']\")"}], "start": 1709458420919, "stop": 1709458421443}], "start": 1709458420919, "stop": 1709458421447}], "start": 1709458420917, "stop": 1709458421448}], "parameters": [{"name": "driver", "value": "'firefox'"}], "start": 1709458420917, "stop": 1709458421449, "uuid": "9dda0ff2-4940-40c8-9c56-abda7a579a22", "historyId": "d0420d77f13b92f03568f1b1a2103168", "testCaseId": "836956ec3053e597c959ce429eb3ce63", "fullName": "tests.test_order_tape.TestOrderTape#test_client_can_find_order_in_history_and_in_tape", "labels": [{"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_order_tape"}, {"name": "subSuite", "value": "TestOrderTape"}, {"name": "host", "value": "MBP-Ilya.cn.ru_novotelecom.ru"}, {"name": "thread", "value": "76774-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_order_tape"}]}